### PR TITLE
feat: Skeleton!

### PR DESCRIPTION
**Pull request by m6jones**
image

https://github.com/BrightspaceHypermediaComponents/foundation-components/pull/56

### CODE REVIEW

File: render/componentFactory.js [hunk 1]
```diff
@@ -39,6 +39,10 @@ class ComponentStore {
 		});
 	}
 
+	skeletonTag() {
+		return this._componentStore.get(defaultType);
```
Comment (m6jones) [inline render/componentFactory.js:43]: Right now it gets the default tag. Seems to be pretty good to just do that. Should be easy to change if we have too.

**Review by tenbradley — COMMENTED** [review]
Overall, this looks good. My only concern is what if there is nothing to load? For instance, looking at an activity without an image or a title results in it loading indefinitely.

But overall, it seems to work. Huzzah.

File: state/HypermediaState.js [hunk 2]
```diff
@@ -38,22 +40,38 @@ class HypermediaState extends Fetchable(Object) {
...
 	 */
 	async allFetchesComplete(skipTheseStates) {
+		await this._waitForFirstFetch;
+		await this.fetchStatus.complete;
 		skipTheseStates = Array.isArray(skipTheseStates) ? skipTheseStates : [skipTheseStates];
 		const skipTheseStatesOnNextStep = [ this, ...skipTheseStates ];
-		await this.fetchStatus.complete;
+		const parentPromises = [];
+		this._parents.forEach((_, parent) => {
+			if (!skipTheseStates.includes(parent)) {
```
Comment (zinabat) [inline state/HypermediaState.js:58]: This prevents circular referencing?

File: test/state/HypermediaState.test.js [hunk 3]
```diff
@@ -677,13 +674,42 @@ describe('HypermediaState class', () => {
...
 			const maxDelay = delayFetchAmountMs * depthOfResponses;
+			expect(observer.parentAnswer).to.equal(42);
+			expect(observer.subEntityFirstAnswer).to.equal(43);
+			expect(observer.subEntityFirstSubAnswer).to.equal(44);
+			expect(observer.subEntitySecondAnswer).to.equal(45);
+			expect(observer.primeLinkAnswer).to.equal(46);
 			expect(timeAfterFetch - timeBeforeFetch).to.be.above(maxDelay);
+		});
+
+		it('the child state, should know when the parent and it\'s children have loaded', async() => {
```
Comment (zinabat) [inline test/state/HypermediaState.test.js:685]: Do we have a test for circular references? Would it be worth adding one if not?

**Review by zinabat — APPROVED** [review]
Just one thing to check but tested and it looks good.

File: state/HypermediaState.js (hunk 2 above)
Comment (m6jones) [reply state/HypermediaState.js:58]: Yup :)

File: test/state/HypermediaState.test.js (hunk 3 above)
Comment (m6jones) [reply test/state/HypermediaState.test.js:685]: I do! There a circular reference in my test data.

**Pull request merged**